sno/checkout.py

```diff
@@ -105,7 +105,6 @@ def checkout(ctx, branch, force, discard_changes, refish):
     reset_wc_if_needed(repo, commit, discard_changes=discard_changes)
 
     repo.set_head(head_ref)
-    repo.reset(commit.oid, pygit2.GIT_RESET_SOFT)
```

Comment:
Why aren't these needed any more?

Reply:
There's nothing to reset that isn't taken care of by `repo.set_head(head_ref)`
- we don't want to modify anything in refs/heads - just HEAD, which set_head takes care of
- we don't have an index
- we don't have a worktree

I think GIT_RESET_SOFT does the right thing here (ie, nothing we didn't already do) but it also checks the index file, which means the index needs to be unlocked. Easier just to get rid of it since it's a no-op.